src/lib/typelib/lib/type.js

```diff
@@ -273,7 +274,32 @@ class Type {
         }
 
         await this.save();
-        await notification.emit(`${this.constructor.typeName}.untagged`, this, tag);
+        await notification.emit(`${this.constructor.typeName}.untagged`, this, tags);
```

Comment:
The thing emitted should be ensureArray(tags). Note that all subscribers to untagged must be updated to handle arrays.
Same thing needs to be done for tagged.